react/components/MadieAlert/index.jsx

```diff
@@ -19,7 +20,42 @@ const MadieAlert = ({
     canClose = true,
     alertProps, // props to pass to outer component
     closeButtonProps, // props to pass to close button
+    copyButton
+
 }) => {
+    const [copyText, setCopyText] = useState("");
+
+    const copyButtonBuilder = (content) => {
+        const traversal = (contentNode, parentNode = true) => {
```

Comment:
I'm wondering whether this logic should live outside the component library. Most of the other components I checked that use hooks do to to manage the display state, but this feels more like business logic.